### PR TITLE
Modifying modes of migration in HarbourBridge

### PR DESCRIPTION
**Pull request by shreyakhajanchi**
1. Added dry run mode of migration
2. Modified schema only and data only mode of migration
3. UI fixes to handle the updated structure
4. Modified the unit and integration tests

### CODE REVIEW

File: cmd/utils.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 Google LLC
```
Comment (manitgupta) [inline cmd/utils.go:1]: We already have a `common/utils/utils.go` class, can we rename this and place this outside the `cmd` folder. The `cmd` folder is the first-level folder for all the sub-commands HB exposes. Let's preserve that.

File: common/utils/utils.go [hunk 2]
```diff
@@ -506,6 +506,60 @@ func ReadSpannerSchema(ctx context.Context, conv *internal.Conv, client *sp.Clie
...
+			spannerCol := spannerTable.ColDefs[col]
+			if colDef.Name != spannerCol.Name || colDef.NotNull != spannerCol.NotNull ||
+				colDef.T.IsArray != spannerCol.T.IsArray || colDef.T.Len != spannerCol.T.Len || colDef.T.Name != spannerCol.T.Name {
+				return fmt.Errorf("column detail for table %v don't match", sessionTable.Name)
+			}
+		}
+		for i := range sessionTable.Indexes {
+			sessionDbIndex := sessionTable.Indexes[i]
+			found := 0
+			for j := range spannerTable.Indexes {
```
Comment (manitgupta) [inline common/utils/utils.go:538]: I am not sure if it is a `golang` thing, but can we use replace existing variables `t, i, j` with something more descriptive? 
It is hard to read the code where they are being referred to at once.

**Review by manitgupta — COMMENTED** [review]
It will be great if we can also document in an open issue what error was not captured in the unit and integration tests we have which led to reverting the PR last time.
We can come back to it later.

File: cmd/utils.go (hunk 1 above)
Comment (shreyakhajanchi) [reply cmd/utils.go:1]: Putting these methods in utils/utils.go was creating an import cycle

**Review by manitgupta — APPROVED** [review]
LGTM

**Pull request merged**